rider/src/main/kotlin/com/jetbrains/rider/UnityReferenceDiscoverer.kt

```diff
@@ -50,3 +53,18 @@ class UnityReferenceDiscoverer(project: Project) : LifetimedProjectComponent(pro
         }
     }
 }
+
+fun Project.isUnityProject(): Boolean {
+    val component = this.getComponent<UnityReferenceDiscoverer>()
+    return component.isUnityProject
+}
+
+fun Project.isConnectedToEditor(): Boolean {
+    val component = this.getComponent<UnityHost>()
+    return component.sessionInitialized.value
+}
+
+fun Project.isConnectedToEditorLive(): Property<Boolean> {
```

Comment:
When Unity connection disappears, I can't see drop-down is changed anyhow.

Reply:
I was thinking about this, but decided not to advise for such a live property. I don't want this drop down to blink on every autosave (refresh in Unity).